pkg/apiserver/profiling/pprof.go

```diff
@@ -3,170 +3,67 @@
-	return tmpPath, nil
-}
-
-type flagSet struct {
-	*flag.FlagSet
-	args []string
-}
-
-func fetchPprof(op *pprofOptions, format string) (string, error) {
-	tmpfile, err := ioutil.TempFile("", op.fileNameWithoutExt)
+	fetcher := &fetcher{profileFetcher: op.fetcher, target: op.target}
+	profileOutputType, err := fetcher.FetchAndWriteToFile(op.duration, tmpPath)
 	if err != nil {
-		return "", fmt.Errorf("failed to create temp file: %v", err)
-	}
-	defer tmpfile.Close() // #nosec
-	tmpPath := fmt.Sprintf("%s.%s", tmpfile.Name(), format)
-	format = "-" + format
-	args := []string{
-		format,
-		// prevent printing stdout
-		"-output", "dummy",
-		"-seconds", strconv.Itoa(int(op.duration)),
+		return "", "", fmt.Errorf("failed to fetch annd write to temp file: %v", err)
 	}
-	address := fmt.Sprintf("%s:%d", op.target.IP, op.target.Port)
-	args = append(args, address)
-	f := &flagSet{
-		FlagSet: flag.NewFlagSet("pprof", flag.PanicOnError),
-		args:    args,
-	}
-	if err := driver.PProf(&driver.Options{
-		Fetch:   &fetcher{profileFetcher: op.fetcher, target: op.target},
-		Flagset: f,
-		UI:      &blankPprofUI{},
-		Writer:  &oswriter{output: tmpPath},
-	}); err != nil {
-		return "", fmt.Errorf("failed to generate profile report: %v", err)
-	}
-	return tmpPath, nil
-}
 
-func (f *flagSet) StringList(o, d, c string) *[]*string {
-	return &[]*string{f.String(o, d, c)}
-}
-
-func (f *flagSet) ExtraUsage() string {
-	return ""
-}
-
-func (f *flagSet) Parse(usage func()) []string {
-	f.Usage = usage
-	_ = f.FlagSet.Parse(f.args)
-	return f.Args()
-}
-
-func (f *flagSet) AddExtraUsage(eu string) {}
-
-// oswriter implements the Writer interface using a regular file.
-type oswriter struct {
-	output string
-}
-
-func (o *oswriter) Open(name string) (io.WriteCloser, error) {
-	f, err := os.Create(o.output)
-	return f, err
+	return tmpPath, profileOutputType, nil
 }
 
 type fetcher struct {
 	target         *model.RequestTargetNode
 	profileFetcher *profileFetcher
 }
 
-func (f *fetcher) Fetch(src string, duration, timeout time.Duration) (*profile.Profile, string, error) {
-	secs := strconv.Itoa(int(duration / time.Second))
+func (f *fetcher) FetchAndWriteToFile(duration uint, tmpPath string) (string, error) {
+	secs := strconv.Itoa(int(duration))
 	url := "/debug/pprof/profile?seconds=" + secs
 
 	resp, err := (*f.profileFetcher).fetch(&fetchOptions{ip: f.target.IP, port: f.target.Port, path: url})
 	if err != nil {
-		return nil, url, err
+		return "", fmt.Errorf("failed to fetch profile with proto format: %v", err)
 	}
 
-	p, err := profile.ParseData(resp)
-	return p, url, err
-}
-
-// blankPprofUI is used to eliminate the pprof logs.
-type blankPprofUI struct{}
-
-func (b blankPprofUI) ReadLine(prompt string) (string, error) {
-	panic("not support")
-}
-
-func (b blankPprofUI) Print(i ...interface{}) {
-}
-
-func (b blankPprofUI) PrintErr(i ...interface{}) {
-}
-
-func (b blankPprofUI) IsTerminal() bool {
-	return false
-}
+	w, err := os.Create(tmpPath)
```

Comment:
You can create a temp file in this function, this is a simplified version if you just want a temp file for writing:

```go
f, err := os.CreateTemp("", op.fileNameWithoutExt)
if err != nil {
  ..
}
_, err = f.Write(resp)
..

_ = f.Close()
```